### PR TITLE
TL/MLX5: enhance reliability protocol in Mcast

### PR DESCRIPTION
**Pull request by MamziB**
TL/MLX5: enhance reliability protocol in Mcast

### CODE REVIEW

File: src/components/tl/mlx5/mcast/tl_mlx5_mcast_coll.c [hunk 1]
```diff
@@ -138,11 +137,13 @@ static inline ucc_status_t ucc_tl_mlx5_mcast_do_bcast(ucc_tl_mlx5_mcast_coll_req
             }
         }
 
-        /* This function will check if we have to do a round of reliability protocol */
-        status = ucc_tl_mlx5_mcast_r_window_recycle(comm, req);
-        if (UCC_OK != status) {
-            return status;
-        }
+
```
Comment (samnordmann) [inline src/components/tl/mlx5/mcast/tl_mlx5_mcast_coll.c:140]: it seems there is extra blank line here, right ?

**Review by samnordmann — APPROVED** [review]
LGTM!

**Review by Sergei-Lebedev — APPROVED** [review]

File: src/components/tl/mlx5/mcast/tl_mlx5_mcast.h [hunk 2]
```diff
@@ -308,10 +308,11 @@ typedef struct ucc_tl_mlx5_mcast_coll_req {
 } ucc_tl_mlx5_mcast_coll_req_t;
 
 typedef struct ucc_tl_mlx5_mcast_oob_p2p_context {
-    ucc_context_h base_ctx;
-    ucc_team_h    base_team;
-    ucc_rank_t    my_team_rank;
-    ucc_subset_t  subset;
+    ucc_context_h   base_ctx;
```
[the remark this answers is not in the document]
Comment (MamziB) [reply src/components/tl/mlx5/mcast/tl_mlx5_mcast.h:311]: it is a good comment, thanks for the suggestion. I will make a new PR for it.

File: src/components/tl/mlx5/mcast/tl_mlx5_mcast_progress.c [hunk 3]
```diff
@@ -10,15 +10,38 @@ static ucc_status_t ucc_tl_mlx5_mcast_recv_completion(ucc_tl_mlx5_mcast_p2p_comp
...
+    ucc_tl_mlx5_mcast_coll_comm_t *comm   = (ucc_tl_mlx5_mcast_coll_comm_t*)comp_obj->data[0];
+    unsigned int                   pkt_id = comp_obj->data[2];
+    struct packet                 *p      = (struct packet *)comp_obj->data[1];
+
+    if (p != NULL) {
+        /* it was a nack packet to our parent */
+        ucc_free(p);
+    }
+
+    if (pkt_id != UINT_MAX) {
```
[the remark this answers is not in the document]
Comment (MamziB) [reply src/components/tl/mlx5/mcast/tl_mlx5_mcast_progress.c:25]: sure I will make a new PR for this.

**Review by janjust — APPROVED** [review]

**Pull request merged**